commit-transaction.md

```diff
@@ -1,27 +1,55 @@
 ---
 title: COMMIT
-summary: Use the COMMIT statement to commit a transaction.
-toc: true
+summary: Commit a transaction with the COMMIT statement in CockroachDB.
+toc: false
 ---
 
-## Description
+The `COMMIT` statement commits the current [transaction](transactions.html).
+
+If the transaction contends for the same resources as other transactions and "loses" when it commits, you can [retry the transaction on the client side](transactions.html#transaction-retries) using [`SAVEPOINT`](savepoint.html) (shown in the [example below](#example)).
+
+If the transaction generates any other types of errors, committing it causes a [rollback](rollback-transaction.html), which discards *all* updates made by statements in the transaction.
+
+<div id="toc"></div>
 
 ## Synopsis
 
 {% include sql/diagrams/commit_transaction.html %}
 
 ## Required Privileges
 
-No [privileges](privileges.html) are required to commit a transaction. 
+No [privileges](privileges.html) are required to commit a transaction. However, privileges are required for each statement within a transaction.
+
+## Aliases
+
+In CockroachDB, `END` is an alias for the `COMMIT` statement.
+
+## Example
+
+### Commit the Current Transaction
+
+After you `BEGIN` a transaction and identify its statements, you must `COMMIT` it to complete the transaction.
+
+~~~ sql
+> BEGIN;
+
+> SAVEPOINT cockroach_restart;
+
+> UPDATE products SET inventory = 0 WHERE sku = '8675309';
+
+> INSERT INTO orders (customer, status) VALUES (1001, 'new');
+
+> RELEASE SAVEPOINT cockroach_restart;
```

Comment:
If you're using SAVEPOINT, then `RELEASE SAVEPOINT` is the real "commit"; the actual `COMMIT` statement does nothing but prepare the connection for the next transaction (things are structured this way for compatibility with other databases and client frameworks). This is surprising and needs to be spelled out clearly.